### PR TITLE
Start waitlisting applications based on age

### LINKED ISSUE #703: Adults in the Slack
Screenshot 2023-02-15 at 16 50 06

Currently, we don't check to see if someone is an adult. That said, if they select "Tertiary education", we flag them as an adult in the Slack. We're just a community for high schoolers, so let's actually alert the user here that adults & college-aged students aren't allowed in the Slack.

### PR DESCRIPTION
**Pull request by maxwofford**
This flashes a warning to anyone who says they're 18+ joining the slack. Should fix #703.

Originally this PR also put them on a waitlist, but after discussion with @karamassie we're just going to flash the warning for now and see what happens.

### CODE REVIEW

**Comment by lachlanjc** [comment]
#### 📝 Changed routes:



- [**/api/join**](https://site-oka1ind57.hackclub.dev/api/join)



Commit 5bef81f0a8a18101ce1e928bf69f0fb62d6f093c (https://site-oka1ind57.hackclub.dev).

**Comment by karamassie** [comment]
we need a line break on the warning message or something? it expanded the view to my whole screen.

**Comment by maxwofford** [comment]
@karamassie just reported this didn't invite her to slack-- I may have broken that logic.

File: components/slack/join-form.js [hunk 1]
```diff
@@ -13,6 +13,9 @@ const JoinForm = ({ sx = {} }) => {
       : { reason: router.query.reason }
   })
 
+  const isAdult = useField('educationLevel').value == 'tertiary'
+  const useWaitlist = process.env.NEXT_PUBLIC_OPEN == 'true'
```
Comment (sampoder) [inline components/slack/join-form.js]:
```suggestion
  const useWaitlist = process.env.NEXT_PUBLIC_OPEN != 'true'
```

I think we need to flip this over. The semantics aren't great but from memory when the Slack isn't "open" we use the waitlist

File: components/slack/join-form.js [hunk 2]
```diff
@@ -79,19 +82,28 @@ const JoinForm = ({ sx = {} }) => {
             required
           />
         </Label>
+        {isAdult && (
+          <Text
+            variant="caption"
+            color="secondary">
```
Comment (sampoder) [inline components/slack/join-form.js]:
```suggestion
            color="secondary"
            as="div"
            sx={{ maxWidth: "600px", textAlign: 'center', mb: 2 }}
          >
```

File: components/slack/join-form.js [hunk 3]
```diff
@@ -79,19 +82,28 @@ const JoinForm = ({ sx = {} }) => {
...
+              Hold your horses! Our Slack community for teenagers, and as such
```
Comment (sampoder) [inline components/slack/join-form.js]:
```suggestion
              Hold your horses! <b>Our Slack community for teenagers</b>, and as such
```

File: components/slack/join-form.js [hunk 4]
```diff
@@ -79,19 +82,28 @@ const JoinForm = ({ sx = {} }) => {
...
+              we're really careful about letting adults join. If you feel you'd
+              have a place here, reach out to conduct@hackclub.com.
```
Comment (sampoder) [inline components/slack/join-form.js]:
```suggestion
              have a place here, reach out to <Link href="mailto:conduct@hackclub.com">conduct@hackclub.com</Link>.
```

File: components/slack/join-form.js [hunk 5]
```diff
@@ -1,4 +1,4 @@
-import { Card, Label, Input, Textarea, Select, Grid } from 'theme-ui'
+import { Card, Label, Input, Textarea, Select, Grid, Text } from 'theme-ui'
```
Comment (sampoder) [inline components/slack/join-form.js]:
```suggestion
import { Card, Label, Input, Textarea, Select, Grid, Text, Link } from 'theme-ui'
```

File: components/slack/join-form.js (hunk 4 above)
Comment (sampoder) [reply components/slack/join-form.js]: It feels a bit weird to have people reaching out to conduct@hackclub.com, "conduct" implies they've done something wrong

**Pull request merged**